contracts/p1/mixins/Trading.sol

```diff
@@ -108,15 +120,27 @@ abstract contract TradingP1 is Multicall, ComponentP1, ReentrancyGuardUpgradeabl
     // This is reentrancy-safe because we're using the `nonReentrant` modifier on every method of
     // this contract that changes state this function refers to.
     // slither-disable-next-line reentrancy-vulnerabilities-1
-    function tryTrade(TradeRequest memory req) internal nonReentrant {
+    function tryTrade(TradeRequest memory req, TradeKind kind) internal nonReentrant {
         /*  */
         IERC20 sell = req.sell.erc20();
         assert(address(trades[sell]) == address(0));
 
         IERC20Upgradeable(address(sell)).safeApprove(address(broker), 0);
         IERC20Upgradeable(address(sell)).safeApprove(address(broker), req.sellAmount);
-        ITrade trade = broker.openTrade(req);
 
+        // Only start the next auction back-to-back if msgSender is self
+        if (_msgSender() != address(this)) {
+            // Require at least 1 empty block between auctions of the same kind
+            // This gives space for someone to start one of the opposite kinds of auctions
+            if (kind == TradeKind.DUTCH_AUCTION) {
+                require(block.number > lastSettlement[TradeKind.DUTCH_AUCTION] + 1, "wait 1 block");
```

Comment:
i thought we wanted to wait 1 full length of an auction?

Reply:
Realized it was unnecessary. Let me know if you think it is necessary. It would mean slower rebalancing.

Reply:
i guess it's really only necessary during an awkward transition phase where this is still semi-manual but bots are coming online...probably not necessary to build specifically for that temporary case.